pkg/services/ssosettings/database/database.go

```diff
@@ -72,33 +72,37 @@ func (s *SSOSettingsStore) List(ctx context.Context) ([]*models.SSOSetting, erro
 }
 
 func (s *SSOSettingsStore) Upsert(ctx context.Context, provider string, data map[string]interface{}) error {
-	err := s.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
-		var err error
-		found, err := sess.Where("provider = ? AND is_deleted = ?", provider, s.sqlStore.GetDialect().BooleanStr(false)).Exist(&models.SSOSetting{})
-
+	return s.sqlStore.WithDbSession(ctx, func(sess *db.Session) error {
+		existing := &models.SSOSetting{
+			Provider:  provider,
+			IsDeleted: false,
+		}
+		found, err := sess.UseBool("is_deleted").Exist(existing)
 		if err != nil {
 			return err
 		}
 
+		now := time.Now().UTC()
```

Comment:
Could you please create a var for timeNow() and use that in the tests as well? 
Like this: https://github.com/grafana/grafana/blob/main/pkg/api/pluginproxy/token_provider.go#L11

Reply:
I'm not very sure what's the value in doing that. Can you please give more details on why do it?

Reply:
Makes it easier to mock `time` (example: https://github.com/grafana/grafana/blob/main/pkg/api/pluginproxy/token_provider_test.go#L76), then you can replace `require.WithinDuration` with value assertion in your tests.

Reply:
Ahh okay, it's for making accurate assertions for time values. I'll update the tests to use this pattern.

Reply:
I made the changes.

I can say that I liked more the previous version that was using `require.WithinDuration` to make the time assertions because of its simplicity and ease of use.

I think that mocking the time.Now() is a bit complicated and error prone. It's easy to introduce a flaky test by typing `time.Now` instead of `timeNow` in a test. Also when a timestamp goes to the database and back then the milliseconds part is lost, that's why I compare the timestamps using the RFC3339 format (it doesn't contain the ms part). This adds complexity with the benefit of assuring the timestamp are properly saved in DB at seconds accuracy.

@mgyongyosi what do you think? Let me know if you see a more simple solution.

Reply:
This solution is fine.

> I think that mocking the time.Now() is a bit complicated and error prone. It's easy to introduce a flaky test by typing time.Now instead of timeNow in a test

Yep, this can happen, but it could be prevented by specifying the time explicitly: 
https://github.com/grafana/grafana/blame/11d4f604f5b11c2f323ac9a34acadc47cc66be5a/pkg/services/authn/clients/ext_jwt_test.go#L495

> This adds complexity with the benefit of assuring the timestamp are properly saved in DB at seconds accuracy.

👍